### PR TITLE
Buddy login functionality

### PR DESCRIPTION
**Pull request by sonaliTekdi**

### CODE REVIEW

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/buddy-login/buddy-login.component.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    private router: Router,
+    private location: Location) { }
+
+  ngOnInit() {
+    this.telemetryService.impression('Login', '/login', 'ET');
+  }
+
+  loginAsBuddy() {
+    this.telemetryService.interact('Submit', 'BuddyLogin', 'NT');
+    if(this.userService.getUser().emis_username===this.email)
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/buddy-login/buddy-login.component.ts:38]: Added space

File: src/app/telemetry.service.ts [hunk 2]
```diff
@@ -191,14 +191,20 @@ export class TelemetryService {
   }
 
   private getEventOptions() {
+    // localStorage.setItem("UID_1",this.userService.getUser().emis_username)
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/telemetry.service.ts:194]: removed

File: src/app/buddy-login/buddy-login.component.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        } else {
+          this.telemetryService.error(
+            data?.message,
+            {
+              err: data?.message,
+              errtype: data?.status,
+            },
+            'DT'
+          );
+          // alert(data.message)
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/buddy-login/buddy-login.component.ts:70]: Removed comments

File: src/app/user/user.service.ts [hunk 4]
```diff
@@ -5,6 +5,8 @@ import jwt_decode from 'jwt-decode';
 })
 export class UserService {
   userData: any;
+  buddyUserData1: any;
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/user/user.service.ts:8]: I have changes two variable into one variable buddyUserData for stored and return token2

File: src/app/user/user.service.ts [hunk 5]
```diff
@@ -17,4 +19,16 @@ export class UserService {
...
     }
   }
-}
+
+  getBuddyUser() {
+    // const token1 = localStorage.getItem('token');
+    const token2 = localStorage.getItem('buddyToken')
+    if (!!token2) {
+      // this.buddyUserData1 = jwt_decode(token1);
+      this.buddyUserData2 = jwt_decode(token2);
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/user/user.service.ts:28]: I have added variable as buddyUserData

File: src/app/telemetry.service.ts [hunk 6]
```diff
@@ -191,14 +191,20 @@ export class TelemetryService {
...
+  const isBuddyLogin = localStorage.getItem('BuddyLogin')
     return {
       object: {},
       context: {
         channel: this.context.channel,
         pdata: this.context.pdata,
         env: 'languagelab.portal',
         sid: this.context.sid,
-        uid: this.userService.getUser().emis_username || 'anonymous',
+        uid: isBuddyLogin==='true'? this.userService.getUser().emis_username+" / "+this.userService.getBuddyUser().emis_username : this.userService.getUser().emis_username || 'anonymous',
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/telemetry.service.ts:205]: Done

File: src/app/auth.service.ts [hunk 7]
```diff
@@ -13,13 +13,21 @@ export class AuthService {
...
+  }
+
+  buddyLogin(email: string, password: string): Observable<any> {
     const credentials = {records: { emis_username: email, emis_password: password }};
     return this.http.post(this.apiUrl, credentials);
   }
 
   logout() {
     localStorage.removeItem('token');
+    localStorage.removeItem('buddyToken');
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/auth.service.ts:28]: Removed

File: src/app/auth.service.ts [hunk 8]
```diff
@@ -13,13 +13,21 @@ export class AuthService {
...
     localStorage.removeItem('guestUser');
+    localStorage.removeItem('BuddyLogin');
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/auth.service.ts:30]: Removed dublicate

File: src/app/auth.service.ts [hunk 9]
```diff
@@ -13,13 +13,21 @@ export class AuthService {
   constructor(private router: Router, private http: HttpClient) { }
 
   login(email: string, password: string): Observable<any> {
+    sessionStorage.setItem('myId',email)
+    const credentials = {records: { emis_username: email, emis_password: password }};
+    return this.http.post(this.apiUrl, credentials);
+  }
+
+  buddyLogin(email: string, password: string): Observable<any> {
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/auth.service.ts:21]: Removed dublicate code

File: src/app/buddy-login/buddy-login.component.ts [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+  eyeSVG = '../../assets/images/password.svg';
+
+  constructor(public userService: UserService,
+    public telemetryService: TelemetryService,
+    private authService: AuthService,
+    private router: Router,
+    private location: Location) { }
+
+  ngOnInit() {
+    this.telemetryService.impression('Login', '/login', 'ET');
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/buddy-login/buddy-login.component.ts:33]: Added

File: src/app/login/login.component.ts [hunk 11]
```diff
@@ -12,6 +12,7 @@ import { Location } from '@angular/common';
   styleUrls: ['./login.component.scss'],
 })
 export class LoginComponent {
+  isBuddyLogin = localStorage.getItem('BuddyLogin')
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/login/login.component.ts:15]: Removed unwanted localstorage code

File: src/app/telemetry.service.ts [hunk 12]
```diff
@@ -191,14 +191,20 @@ export class TelemetryService {
   }
 
   private getEventOptions() {
+    // localStorage.setItem("UID_1",this.userService.getUser().emis_username)
+    // localStorage.setItem("UID_2",this.userService.getBuddyUser())
+    // console.log(this.userService.getBuddyUser());
+  const isBuddyLogin = localStorage.getItem('BuddyLogin')
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/telemetry.service.ts:197]: We are using it.

File: src/app/auth.service.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/auth.service.ts:28]: If I removed it then buddyToken not  removed form local storage that's why I leave it as it is

File: src/app/auth.service.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/auth.service.ts:30]: BuddyLogin and BuddyToken both are different so I leave it as it is.

File: src/app/login/login.component.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/app/login/login.component.ts:15]: I have set it for to get Buddy login userid and username from localstorage.so I leave as it is

**Pull request merged**